xhistogram/core.py

```diff
@@ -146,6 +147,57 @@ def _histogram_2d_vectorized(
     return bin_counts
 
 
+def _bincount(*all_arrays, weights=False, axis=None, bins=None, density=None):
+
+    # is this necessary?
+    all_arrays_broadcast = broadcast_arrays(*all_arrays)
```

Comment:
If you go with the change suggested in https://github.com/xgcm/xhistogram/pull/49#discussion_r640872588 (no `np.broadcast` in top-level `histogram`), then I think this broadcast _is_ necessary. Otherwise, if `a0` happens to have dimensions that would be added or expanded by broadcasting, then your `kept_axes_shape` calculation could be wrong, and `reshape_input` might move the wrong axes.

However, if you added an `np.broadcast` in top-level `histogram`, then I think you could remove this one here.

Reply:
So with this PR, none of this code path will ever see dask arrays. So we can probably remove the duck_array_ops module.